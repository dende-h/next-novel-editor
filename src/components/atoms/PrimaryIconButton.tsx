import { IconButton, IconButtonProps } from "@chakra-ui/react";
import { memo, VFC } from "react";

type Props = IconButtonProps;

export const PrimaryIconButton: VFC<Props> = memo((props) => {
	const { ...IconButtonProps } = props;
	return (
		<>
			<IconButton {...IconButtonProps} />
		</>
	);
});

PrimaryIconButton.displayName = "PrimaryIconButton"