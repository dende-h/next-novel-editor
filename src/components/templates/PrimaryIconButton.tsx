import { IconButton, IconButtonProps } from "@chakra-ui/react";

type Props = {
	icon: JSX.Element;
	defaultColor?: string;
	changeColor?: string;
	bgColor?: string;
	focusOutline?: string;
	isDisableHoverAnimation?: boolean;
} & IconButtonProps;

export const PrimaryIconButton = (props: Props) => {
	const { icon, defaultColor, changeColor, bgColor, focusOutline, isDisableHoverAnimation, ...IconButtonProps } = props;

	return (
		<>
			<IconButton
				{...IconButtonProps}
				transitionProperty="all"
				transitionDuration="0.5s"
				transitionTimingFunction={"ease-out"}
				_hover={isDisableHoverAnimation ? { color: defaultColor } : { color: changeColor, fontSize: "24px" }}
				_focus={{ boxShadow: focusOutline }}
				icon={icon}
				color={defaultColor}
				backgroundColor={bgColor}
				border={"none"}
				borderRadius={"full"}
			/>
		</>
	);
};
