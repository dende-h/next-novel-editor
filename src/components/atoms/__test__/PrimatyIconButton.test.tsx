import { render } from "@testing-library/react";
import { PrimaryIconButton } from "../PrimaryIconButton";
import "@testing-library/jest-dom";

describe("PrimaryIconButton", () => {
	it("renders without crashing", () => {
		const { container } = render(<PrimaryIconButton aria-label="Primary Button" />);
		expect(container.firstChild).toMatchSnapshot();
	});
});
