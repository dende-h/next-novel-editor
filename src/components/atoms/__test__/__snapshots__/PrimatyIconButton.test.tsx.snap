// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryIconButton renders without crashing 1`] = `
<button
  aria-label="Primary Button"
  class="chakra-button css-198bfjk"
  type="button"
/>
`;
